packages/p2p-v2/src/components/PaymentMethodField/PaymentMethodField.tsx

```diff
@@ -0,0 +1,84 @@
+ * @param {string} props.field - The name of the field
+ * @param {boolean} props.required - Whether the field is required or not
+ * @returns {JSX.Element}
+ * @example <PaymentMethodField control={control} defaultValue={defaultValue} displayName={displayName} field={field} required={required} />
+ * **/
+const PaymentMethodField = ({ control, defaultValue, displayName, field, required }: TPaymentMethodField) => {
+    return (
+        <div className='p2p-v2-payment-method-form__field-wrapper'>
+            {field === 'instructions' ? (
+                <Controller
+                    control={control}
+                    defaultValue={defaultValue}
+                    name={field}
+                    render={({ field: { onBlur, onChange, value }, fieldState: { error } }) => {
+                        return (
+                            <TextArea
+                                hint={error?.message}
+                                isInvalid={!!error?.message}
+                                label={displayName}
+                                onBlur={onBlur}
+                                onChange={onChange}
+                                value={value}
+                            />
+                        );
+                    }}
+                    rules={{
+                        pattern: {
+                            message: `${displayName} can only include letters, numbers, spaces, and any of these symbols: -+.,'#@():;`, // TODO: Remember to translate this
+                            value: VALID_SYMBOLS_PATTERN,
+                        },
+                        required: required ? 'This field is required.' : false,
+                    }}
+                />
+            ) : (
+                <Controller
+                    control={control}
+                    defaultValue={defaultValue}
+                    name={field}
+                    render={({ field: { onBlur, onChange, value }, fieldState: { error } }) => {
+                        return (
+                            <TextField
+                                errorMessage={error?.message}
+                                isInvalid={!!error?.message}
+                                label={displayName}
+                                onBlur={onBlur}
+                                onChange={onChange}
+                                value={value}
+                            />
+                        );
+                    }}
+                    rules={{
+                        pattern: {
+                            message: `${displayName} can only include letters, numbers, spaces, and any of these symbols: -+.,'#@():;`, // TODO: Remember to translate this
+                            value: VALID_SYMBOLS_PATTERN,
+                        },
+                        required: required ? 'This field is required.' : false,
+                    }}
+                />
+            )}
```

Comment:
would something like this work? cause the only difference is the TextArea/Field being rendered
```suggestion
                <Controller
                    control={control}
                    defaultValue={defaultValue}
                    name={field}
                    render={({ field: { onBlur, onChange, value }, fieldState: { error } }) => {
                        return (
                          {field === 'instructions' ? 
                              <TextArea
                                  hint={error?.message}
                                  isInvalid={!!error?.message}
                                  label={displayName}
                                  onBlur={onBlur}
                                  onChange={onChange}
                                  value={value}
                              /> :
                              <TextField
                                  errorMessage={error?.message}
                                  isInvalid={!!error?.message}
                                  label={displayName}
                                  onBlur={onBlur}
                                  onChange={onChange}
                                  value={value}
                              />
                            }
                        );
                    }}
                    rules={{
                        pattern: {
                            message: `${displayName} can only include letters, numbers, spaces, and any of these symbols: -+.,'#@():;`, // TODO: Remember to translate this
                            value: VALID_SYMBOLS_PATTERN,
                        },
                        required: required ? 'This field is required.' : false,
                    }}
                />
```